sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/AbstractConfiguration.java

```diff
@@ -19,68 +19,73 @@
   }
 
-  public abstract String getRoslynJsonReportPathProperty();
+  protected abstract String getRoslynJsonReportPathProperty();
 
-  public abstract String getAnalyzerWorkDirProperty();
+  protected abstract String getAnalyzerWorkDirProperty();
 
-  public abstract String getAnalyzerReportDir();
+  protected abstract String getAnalyzerReportDir();
 
-  public boolean isReportsComingFromMSBuild() {
-    if (reportsComingFromMSBuild == null) {
-      reportsComingFromMSBuild = areProtobufReportsPresent();
-    }
-    return reportsComingFromMSBuild;
-  }
-
-  private boolean areProtobufReportsPresent() {
-    if (!settings.hasKey(getAnalyzerWorkDirProperty())) {
+  public boolean areProtobufReportsPresent() {
+    if (!configuration.hasKey(getAnalyzerWorkDirProperty())) {
+      LOG.warn("Property missing: '" + getAnalyzerWorkDirProperty() + "'. No protobuf files will be loaded.");
       return false;
     }
-    Path analyzerOutputDir = protobufReportPathFromScanner();
+    Path analyzerOutputDir = protobufReportPath();
 
     if (!analyzerOutputDir.toFile().exists()) {
-      LOG.info("Analyzer working directory does not exist");
+      LOG.warn("Analyzer working directory does not exist: " + analyzerOutputDir.toAbsolutePath() + ". No protobuf files will be loaded.");
       return false;
     }
 
-    try (DirectoryStream<Path> files = Files.newDirectoryStream(analyzerOutputDir, p -> p.toAbsolutePath().toString().toLowerCase().endsWith(".pb"))) {
+    try (DirectoryStream<Path> files = Files.newDirectoryStream(analyzerOutputDir, p -> p.getFileName().toString().toLowerCase().endsWith(".pb"))) {
       long count = StreamSupport.stream(files.spliterator(), false).count();
+      if (count == 0) {
+        LOG.warn("Analyzer working directory contains no .pb file(s). No protobuf files will be loaded.");
+        return false;
+      }
       LOG.info("Analyzer working directory contains " + count + " .pb file(s)");
-      return count != 0;
+      return true;
     } catch (IOException e) {
-      LOG.warn("Could not check for .pb files in " + analyzerOutputDir.toAbsolutePath().toString(), e);
-      return false;
+      throw new IllegalStateException("Could not check for .pb files in " + analyzerOutputDir.toAbsolutePath(), e);
     }
   }
 
-  public Path protobufReportPathFromScanner() {
-    String analyzerWorkDirPath = settings.getString(getAnalyzerWorkDirProperty());
+  public boolean isRoslynReportPresent() {
+    return configuration.hasKey(getRoslynJsonReportPathProperty());
+  }
+
+  public Path roslynReportPath() {
+    return Paths.get(configuration.get(getRoslynJsonReportPathProperty()).get());
+  }
+
+  /**
+   * Returns path of the directory containing all protobuf files.
+   * Check if it exists with {@link AbstractConfiguration#areProtobufReportsPresent()}.
+   */
+  public Path protobufReportPath() {
+    String analyzerWorkDirPath = configuration.get(getAnalyzerWorkDirProperty()).get();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3655)